src/rpc/txoutproof.cpp

```diff
@@ -127,42 +180,140 @@ static RPCHelpMan verifytxoutproof()
         "and throwing an RPC error if the block is not in our best chain\n",
         {
             {"proof", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The hex-encoded proof generated by gettxoutproof"},
+            {"options", RPCArg::Type::OBJ_NAMED_PARAMS, RPCArg::Optional::OMITTED, "",
+                {
+                    {"verify_witness", RPCArg::Type::BOOL, RPCArg::Default{false}, "If true, also verifies the associated wtxid/hash of the specified transactions (if included in proof)"},
+                },
+            },
         },
+        {
         RPCResult{
+            "If verify_witness is false or unspecified",
             RPCResult::Type::ARR, "", "",
             {
                 {RPCResult::Type::STR_HEX, "txid", "The txid(s) which the proof commits to, or empty array if the proof cannot be validated."},
             }
+        },
+            RPCResult{
+                "If verify_witness is true and the proof valid", RPCResult::Type::OBJ, "", "",
+                {
+                    {RPCResult::Type::STR_HEX, "blockhash", "The block hash this proof links to"},
+                    {RPCResult::Type::NUM, "blockheight", "The height of the block this proof links to"},
+                    {RPCResult::Type::NUM, "confirmations", /*optional=*/true, "Number of blocks (including the one with the transactions) confirming these transactions"},
+                    {RPCResult::Type::NUM, "confirmations_assumed", /*optional=*/true, "The number of unverified blocks confirming these transactions (eg, in an assumed-valid UTXO set)"},
```

Comment:
This information (height, confirmations, assumeutxo info) seems better obtained via `getblockheader` on the blockhash to me.